File: examples/java/Runner.java
// Example Gota runner (Java): FNV-1a over the buffer. See ../README.md.
//   javac Gota.java Runner.java && java Runner [buffer_bytes] [warmup_s] [measure_s]
public final class Runner {
    public static void main(String[] args) {
        Gota.run("java", args, (b, data) -> {
            b.bench("fnv1a-64", () -> {
                long h = 0xcbf29ce484222325L;
                for (byte x : data) {
                    h = (h ^ (x & 0xffL)) * 0x100000001b3L;
                }
                data[0] ^= (byte) h; // sink: consume h
            });
        });
    }
}

File: examples/zig/runner.zig
// Example Gota runner (Zig): FNV-1a over the buffer. See ../README.md.
//   zig build-exe runner.zig -O ReleaseFast && ./runner [buffer_bytes] [warmup_s] [measure_s]
const std = @import("std");
const gota = @import("gota.zig");

fn fnv1a(data: []u8) void {
    var h: u64 = 0xcbf29ce484222325;
    for (data) |x| h = (h ^ @as(u64, x)) *% 0x100000001b3;
    data[0] ^= @truncate(h); // sink: consume h
}

fn register(b: *gota.Bencher) !void {
    try b.bench("fnv1a-64", fnv1a);
}

pub fn main(init: std.process.Init) !void {
    try gota.run(init, "zig", register);
}
